### PR TITLE
Improve IO ban information

### PR DESCRIPTION
**Pull request by ramon-bernardo**
### Pull Request Prelude
- [x] I have followed [proper The Forgotten Server code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed
- Use optional when instance a ban info
- Define "none" when empty reason 

[cont]: https://github.com/otland/forgottenserver/wiki/Contributing
[code]: https://github.com/otland/forgottenserver/wiki/TFS-Coding-Style-Guide

### CODE REVIEW

**Review by EPuncker — APPROVED** [review]

File: src/ban.cpp [hunk 1]
```diff
@@ -44,15 +44,15 @@ bool Ban::acceptConnection(const Connection::Address& clientIP)
 	return true;
 }
 
-bool IOBan::isAccountBanned(uint32_t accountId, BanInfo& banInfo)
+const std::shared_ptr<BanInfo> IOBan::isAccountBanned(uint32_t accountId)
```
Comment (ranisalt) [inline src/ban.cpp:47]: `std::optional<BanInfo>` is a better option, since it will be stack allocated and use less memory.

File: src/ban.cpp [hunk 2]
```diff
@@ -44,15 +44,15 @@ bool Ban::acceptConnection(const Connection::Address& clientIP)
 	return true;
 }
 
-bool IOBan::isAccountBanned(uint32_t accountId, BanInfo& banInfo)
+const std::optional<BanInfo> IOBan::isAccountBanned(uint32_t accountId)
```
Comment (MillhioreBT) [inline src/ban.cpp:47]: The methods should be called: `getMethodName` instead of `isMethodName`.

**Review by MillhioreBT — CHANGES_REQUESTED** [review]

File: src/ban.cpp (hunk 2 above)
Comment (ramon-bernardo) [reply src/ban.cpp]: _has_ may be better than _get_

Comment (EPuncker) [reply src/ban.cpp]: both ugly 😄

Comment (Shawak) [reply src/ban.cpp]: Why not call them `getAccountBanInfo` and `getIPBanInfo`?

**Review by EPuncker — APPROVED** [review]

**Review by EPuncker — APPROVED** [review]

**Review by MillhioreBT — APPROVED** [review]

**Pull request merged**